src/rtc_time.cpp

```diff
@@ -7,84 +7,117 @@
 
 #include "rtc_time.hpp"
 
-uint8_t RTCTime::getSeconds() {
+uint8_t RTCTime::getSeconds() const {
     return seconds;
 }
 
-uint8_t RTCTime::getMinutes() {
+uint8_t RTCTime::getMinutes() const {
     return minutes;
 }
 
-uint8_t RTCTime::getHours() {
+uint8_t RTCTime::getHours() const {
     return hours;
 }
 
-uint8_t RTCTime::getDayOfTheWeek() {
+uint8_t RTCTime::getDayOfTheWeek() const {
     return dayOfTheWeek;
 }
 
-uint8_t RTCTime::getDayOfTheMonth() {
+uint8_t RTCTime::getDayOfTheMonth() const {
     return dayOfTheMonth;
 }
 
-uint8_t RTCTime::getMonth() {
+uint8_t RTCTime::getMonth() const {
     return month;
 }
 
-uint8_t RTCTime::getYear() {
+uint8_t RTCTime::getYear() const {
     return year;
 }
 
 
 
 void RTCTime::setSeconds(uint8_t seconds) {
-    this->seconds = seconds;
+    seconds = seconds;
 }
 
 void RTCTime::setMinutes(uint8_t minutes) {
-    this->minutes = minutes;
+    minutes = minutes;
 }
 
 void RTCTime::setHours(uint8_t hours) {
-    this->hours = hours;
+    hours = hours;
 }
 
 void RTCTime::setDayOfTheWeek(uint8_t dayOfTheWeek) {
-    this->dayOfTheWeek = dayOfTheWeek;
+    dayOfTheWeek = dayOfTheWeek;
 }
 
 void RTCTime::setDayOfTheMonth(uint8_t dayOfTheMonth) {
-    this->dayOfTheMonth = dayOfTheMonth;
+    dayOfTheMonth = dayOfTheMonth;
 }
 
 void RTCTime::setMonth(uint8_t month) {
-    this->month = month;
+    month = month;
 }
 
 void RTCTime::setYear(uint8_t year) {
-    this->year = year;
+    year = year;
 }
 
 
 void RTCTime::setTime(uint8_t seconds, uint8_t minutes, uint8_t hours) {
-    this->seconds = seconds;
-    this->minutes = minutes;
-    this->hours = hours;
+    seconds = seconds;
+    minutes = minutes;
+    hours = hours;
 }
 
 void RTCTime::setDate(uint8_t dayOfTheMonth, uint8_t month, uint8_t year) {
-    this->dayOfTheMonth = dayOfTheMonth;
-    this->month = month;
-    this->year = year;
+    dayOfTheMonth = dayOfTheMonth;
+    month = month;
+    year = year;
 }
 
 void RTCTime::set(uint8_t seconds, uint8_t minutes, uint8_t hours, uint8_t dayOfTheWeek,
             uint8_t dayOfTheMonth, uint8_t month, uint8_t year) {
-    this->seconds = seconds;
-    this->minutes = minutes;
-    this->hours = hours;
-    this->dayOfTheWeek = dayOfTheWeek;
-    this->dayOfTheMonth = dayOfTheMonth;
-    this->month = month;
-    this->year = year;
+    seconds = seconds;
+    minutes = minutes;
+    hours = hours;
+    dayOfTheWeek = dayOfTheWeek;
+    dayOfTheMonth = dayOfTheMonth;
+    month = month;
+    year = year;
+}
+
+bool operator< (const RTCTime& lhs, const RTCTime& rhs) {
```

Comment:
o.0

You should be able to do this in a better way, remove those sneaky returns and don't nest that deep